### ISSUE TITLE
Figure out when we need to wait for the new osd node to come up after node replacement process

### ISSUE THREAD

**Issue by yitzhak12** [red-hat-storage/ocs-ci#4550]
After the node replacement process, we have a few verification steps. 
One of the steps is to wait for the new osd node to come up - it seems that in this test https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/1291/testReport/tests.manage.z_cluster.nodes.test_node_replacement_proactive/TestNodeReplacement/test_nodereplacement_proactive/, the new osd node 'compute-7' didn't come up because we already had 6 osd nodes in the cluster. As you can see here, http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/j062vuf1cs36-t4a/j062vuf1cs36-t4a_20210702T075201/logs/failed_testcase_ocs_logs_1625220237/test_nodereplacement_proactive_ocs_logs/ocs_must_gather/quay-io-rhceph-dev-ocs-must-gather-sha256-f0445113e318c8e1e8ffd1f581beab6d7889a34ebfa17862dd2f6ff7958269d4/namespaces/openshift-storage/oc_output/pods_-owide the new osd node 'compute-7' isn't associated with the osd pods. So it seems that there are a few cases, like when we have 6 osd nodes, that we don't need to wait for the new osd node after the node replacement process.

**Comment by yitzhak12** [comment]
To summarize the issue: in this run, we have 7 OCS worker nodes as you can see here http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/j062vuf1cs36-t4a/j062vuf1cs36-t4a_20210702T075201/logs/failed_testcase_ocs_logs_1625220237/test_nodereplacement_proactive_ocs_logs/ocs_must_gather/quay-io-rhceph-dev-ocs-must-gather-sha256-f0445113e318c8e1e8ffd1f581beab6d7889a34ebfa17862dd2f6ff7958269d4/cluster-scoped-resources/oc_output/get_nodes_-o_wide_--show-labels and 15 OSDs. From the "pods -o wide" output we see that all the OSD pods are divided between the 6 OCS worker nodes: 

> rook-ceph-osd-0-57f8db4dfb-2cnpt                                  1/1     Running     0          39m     10.130.2.69    compute-3   <none>           <none>
rook-ceph-osd-1-6487484b67-zggqk                                  1/1     Running     0          5h46m   10.128.2.15    compute-1   <none>           <none>
rook-ceph-osd-10-65f8fd7476-hkmzg                                 1/1     Running     0          93m     10.129.4.44    compute-6   <none>           <none>
rook-ceph-osd-11-c5dcbc665-4dmkt                                  1/1     Running     0          163m    10.130.2.63    compute-3   <none>           <none>
rook-ceph-osd-12-c6d95c48f-gs9fh                                  1/1     Running     0          135m    10.128.4.40    compute-4   <none>           <none>
rook-ceph-osd-13-f799c57fc-7mrbf                                  1/1     Running     0          135m    10.129.4.43    compute-6   <none>           <none>
rook-ceph-osd-14-7cf64d58f-9r49n                                  1/1     Running     0          135m    10.130.2.66    compute-3   <none>           <none>
rook-ceph-osd-2-7b5bfd7985-jlrqh                                  1/1     Running     0          93m     10.131.2.131   compute-5   <none>           <none>
rook-ceph-osd-3-84d55fc8f4-7n8m7                                  1/1     Running     0          39m     10.130.2.71    compute-3   <none>           <none>
rook-ceph-osd-4-7d7fbf5b5f-ckdjj                                  1/1     Running     0          4h33m   10.128.4.33    compute-4   <none>           <none>
rook-ceph-osd-5-98b9df86b-n5kls                                   1/1     Running     0          4h33m   10.129.4.39    compute-6   <none>           <none>
rook-ceph-osd-6-86f7fc7d7-tdh8v                                   1/1     Running     0          3h20m   10.129.4.41    compute-6   <none>           <none>
rook-ceph-osd-7-6f96bf4996-khb8n                                  1/1     Running     0          3h20m   10.128.4.37    compute-4   <none>           <none>
rook-ceph-osd-8-7c7d449ffd-d9c9s                                  1/1     Running     0          39m     10.130.2.70    compute-3   <none>           <none>
rook-ceph-osd-9-69cf476994-q8mf7                                  1/1     Running     0          163m    10.128.2.19    compute-1   <none>           <none>

And the new OCS worker node 'compute-7' isn't associated with any of them. Is it the right behavior, or maybe is it a bug?
@PrasadDesala @prsurve @sidhant-agrawal, what do you think?

**Comment by PrasadDesala** [comment]
@yitzhak12 This test results are on which OCS version?

**Comment by yitzhak12** [comment]
@PrasadDesala From what I see here https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/1291/parameters/, it's on OCS 4.6.6

**Comment by PrasadDesala** [comment]
@yitzhak12 The OSDs are not equally distributed across the hosts till OCS 4.6, so there is no guarantee that OSDs will run on the newly added node. With Topology spread constraints feature in 4.7, the OSDs are evenly spread across the hosts. So this test when ran on >=4.7 we should see OSDs running on the newly added node.

**Comment by yitzhak12** [comment]
I see, thanks @PrasadDesala

**Comment by yitzhak12** [comment]
The error appears again in this Jenkins run: https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/1540/. Failed test: https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/1540/testReport/tests.manage.z_cluster.nodes.test_node_replacement_proactive/TestNodeReplacement/test_nodereplacement_proactive/. Again I looked at the "pods -o wide" output http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/j068vuf1cs36-t4a/j068vuf1cs36-t4a_20210802T125157/logs/failed_testcase_ocs_logs_1627916581/test_nodereplacement_proactive_ocs_logs/ocs_must_gather/quay-io-rhceph-dev-ocs-must-gather-sha256-180ca4c2ca1f8bfd59251ef37dc6f0b0c6f6b651383dad7a34ef67c0374617f5/namespaces/openshift-storage/oc_output/pods_-owide, and we see that all the OSD pods are divided between the 5 OCS old worker nodes: 

> rook-ceph-osd-0-57c8d7d889-zgzh4                                  2/2     Running     0          3h7m    10.129.2.15    compute-0   <none>           <none>
rook-ceph-osd-1-5955964549-5srsd                                  2/2     Running     0          42m     10.131.2.59    compute-1   <none>           <none>
rook-ceph-osd-10-864f59db4c-qs45b                                 2/2     Running     0          156m    10.129.2.16    compute-0   <none>           <none>
rook-ceph-osd-11-56d9d6859c-84h4n                                 2/2     Running     0          156m    10.131.2.54    compute-1   <none>           <none>
rook-ceph-osd-2-5cf4b959-9jq26                                    2/2     Running     0          4h54m   10.131.0.13    compute-4   <none>           <none>
rook-ceph-osd-3-7d7df8c89b-zl9kt                                  2/2     Running     2          3h48m   10.129.2.9     compute-0   <none>           <none>
rook-ceph-osd-4-6499dc785d-lz9jf                                  2/2     Running     0          3h48m   10.131.0.14    compute-4   <none>           <none>
rook-ceph-osd-5-5959d6c86c-k2dnv                                  2/2     Running     0          3h48m   10.131.2.51    compute-1   <none>           <none>
rook-ceph-osd-6-c5f885f97-hplf9                                   2/2     Running     0          3h8m    10.130.2.55    compute-5   <none>           <none>
rook-ceph-osd-7-67f8f778c-2fh72                                   2/2     Running     0          41m     10.131.2.61    compute-1   <none>           <none>
rook-ceph-osd-8-7f9c87495-bhqt9                                   2/2     Running     0          113m    10.129.2.18    compute-0   <none>           <none>
rook-ceph-osd-9-7bb5f5cf84-qklwj                                  2/2     Running     0          157m    10.130.2.72    compute-5   <none>           <none>

And the new OCS worker node 'compute-6' isn't associated with any of them. @PrasadDesala @prsurve is it a bug?

**Comment by yitzhak12** [comment]
OCS registry image: quay.io/rhceph-dev/ocs-registry:4.8.0-175.ci

**Comment by PrasadDesala** [comment]
The new worker node will be taken into account for the OSD distribution only after a new add-capacity operation, as the incoming pods will be distributed based on TSC.

**Comment by yitzhak12** [comment]
I will create a new PR that will solve the issue
